drafts/client-interface.md

```diff
@@ -3,11 +3,150 @@
 This document describes the interface of a yet-to-be-built reference client
 library in pseudo code.
 
+The hydra client works similar to a browser. You point it to a URL. The URL becomes the current resource. 
+Then you interact with the controls of the resource.
 
-## Retrieve a resource
+The examples show how a cli application could use the hydra client for interaction with an API.
+
+## Find API entrypoint for website
+
+Hydra has a special mechanism to discover API entrypoints.
+
+```
+hydra.findEntrypoint(website)
+print hydra.representation 
+// the API entrypoint is now the current representation
+```
+
+## Retrieve a resource from URI
+
+### Using uniform interface
+
+Assuming that apiurl is known as API entrypoint and that it is a 
+[FoodEstablishment](http://schema.org/FoodEstablishment). The menu information is not embedded, but there is only a link to the menu, 
+i.e. the FoodEstablishment representation has a control to retrieve the menu. 
+
+This assumes that the client is aware of the uniform interface of the underlying protocol. 
+In the example below the client knows that it uses the uniform interface of HTTP.
+```
+hydra.setLocation(apiurl);
+// apiurl is now the current representation
+
+// we want to retrieve or download the menu of the food establishment
+menu = hydra.controls["http://schema.org/hasMenu"]
+if(menu && menu.methods["GET"])
+    hydra.execute("GET", menu.href)
```

Comment:
> Or does hasLinkOfType mean that only GET affordances are eligible?

Yes, that was my intention behind that clunky name :-)

> If so, should there be hasPutOfType, hasPostOfType, hasPatchOfType etc.?

I'd abstract it from HTTP. There are safe transitions (i.e., simple links or GET) and unsafe ones (operations - let's squint for a bit on the fact that operations might be safe as well for now)

> Multiple links are a good point. How should we enable the user of the hydraClient to identify the link she is looking for?

That's up to the app but I imagine it would need to look at properties of the link. If there are none, it would likely consider them as equivalent and follow both.